juju/juju
# [JUJU-2350] Retry secret get and delete to account for drained secrets

When the unit agent gets secret content or deletes secret revisions, the active backend may have changed and the secret content moved across between when the hook context was set up and the secrets loaded, and the commit operations taking place. So we need to add retries to the get and delete. The retry involves reloading the revision metadata each time to pick up the new backend reference info, so a new facade api was needed.

For future, we allow for the revisions to be marked as pending delete when the backend info is loaded - this will be used to prevent the content being moved to a new backend while the delete is taking place.

## Checklist


## QA steps

At the moment, secrets aren't drained.
Create a secret in the controller backend and add some revisions.
Set up vault and create some new secrets.
Ensure that fetching secret content works.
Ensure that deleting secret revisions works.
